File: katt3/avian/attachments/java/main.java
import java.util.*;

public class main {

	public static void main(String... args) {
		Scanner sc = new Scanner(System.in);
		int C, K, N;
		String X;
		C = sc.nextInt();
		K = sc.nextInt();
		N = sc.nextInt();
		X = sc.next();
		String[] Y = avian.encode(C, K, N, X);


		if (Y.length != K) {
			System.out.println("Encoder gave " + Y.length + " strings, expected " + K);
			return;
		}
		for (String s : Y) {
			if (s.length() != N) {
				System.out.println("Encoder gave string of length " + s.length() + " strings, expected " + N);
				return;
			}
			for (char c : s.toCharArray()) {
				if (c != '0' && c != '1') {
					System.out.println("Encoder gave string with characted " + c + " strings, expected 0 or 1");
					return;
				}
			}
		}
		int[] I = new int[C];
		for (int i = 0; i < C; i++) {
			I[i] = sc.nextInt();
		}
		String[] subset = new String[C];
		for (int i = 0; i < C; i++) {
			subset[i] = Y[I[i]];
		}
		String dec = avian.decode(C, K, N, subset, I);
		System.out.println(dec);
	}
}

File: katt3/avian/attachments/csharp/Main.cs
using System;

public class main {

	public static int Main(string[] args) {
		int C, K, N;
		string X;
		string[] parts = Console.ReadLine().Split();
		C = Int32.Parse(parts[0]);
		K = Int32.Parse(parts[1]);
		N = Int32.Parse(parts[2]);
		X = Console.ReadLine();

		string[] Y = avian.Encode(C, K, N, X);
		if (Y.Length != K) {
			Console.WriteLine("Encoder gave " + Y.Length + " strings, expected " + K);
			return 0;
		}
		foreach (string s in Y) {
			if (s.Length != N) {
				Console.WriteLine("Encoder gave string of length " + s.Length + " strings, expected " + N);
				return 0;
			}
			foreach (char c in s.ToCharArray()) {
				if (c != '0' && c != '1') {
					Console.WriteLine("Encoder gave string with characted " + c + " strings, expected 0 or 1");
					return 0;
				}
			}
		}
		parts = Console.ReadLine().Split();
		int[] I = new int[C];
		for (int i = 0; i < C; i++) {
			I[i] = Int32.Parse(parts[i]);
		}
		string[] subset = new string[C];
		for (int i = 0; i < C; i++) {
			subset[i] = Y[I[i]];
		}
		string dec = avian.Decode(C, K, N, subset, I);
		Console.WriteLine(dec);
		return 0;
	}
	
}
